src/main/java/org/onebusaway/alexa/AuthedSpeechlet.java

```diff
@@ -0,0 +1,105 @@
+import com.amazon.speech.speechlet.*;
+import com.amazon.speech.ui.PlainTextOutputSpeech;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j;
+import org.onebusaway.alexa.lib.ObaAgencies;
+import org.onebusaway.alexa.lib.ObaUserClient;
+import org.onebusaway.alexa.storage.ObaUserDataItem;
+import org.onebusaway.io.client.elements.ObaArrivalInfo;
+import org.onebusaway.io.client.util.ArrivalInfo;
+
+import javax.annotation.Resource;
+import java.time.Instant;
+
+@NoArgsConstructor
+@Log4j
+public class AuthedSpeechlet implements Speechlet {
+    @Resource
+    private AnonSpeechlet anonSpeechlet;
+
+    @Resource
+    private ObaAgencies obaAgencies;
+
+    private ObaUserClient obaUserClient;
+
+    private ObaUserDataItem userData;
+
+    public void setUserData(ObaUserDataItem userData) {
+        this.userData = userData;
+        obaUserClient = new ObaUserClient(
+                obaAgencies.agencyForCity(userData.getCity()).get());
+    }
+
+    @Override
+    public SpeechletResponse onIntent(final IntentRequest request,
+                                      final Session session)
+                                      throws SpeechletException {
+        Intent intent = request.getIntent();
+        if ("SetCityIntent".equals(intent.getName())) {
+            return anonSpeechlet.onIntent(request, session);
+        } else if ("GetCityIntent".equals(intent.getName())) {
+            PlainTextOutputSpeech out = new PlainTextOutputSpeech();
+            out.setText(
+                    String.format("You live in %s.",
+                            userData.getCity()));
+            return SpeechletResponse.newTellResponse(out);
+        } else if ("SetStopNumberIntent".equals(intent.getName())) {
+            return anonSpeechlet.onIntent(request, session);
+        } else if ("GetArrivalsIntent".equals(intent.getName())) {
+            return tellArrivals();
+        } else {
+            throw new SpeechletException("Did not recognize intent name");
+        }
+    }
+
+    @Override
+    public SpeechletResponse onLaunch(final LaunchRequest request,
+                                      final Session session)
+                                      throws SpeechletException {
+        return tellArrivals();
+    }
+
+    @Override
+    public void onSessionStarted(final SessionStartedRequest request,
+                                 final Session session) {
+
+    }
+
+    @Override
+    public void onSessionEnded(final SessionEndedRequest request,
+                               final Session session) {
+
+    }
+
+    private SpeechletResponse tellArrivals() {
+        ObaArrivalInfo[] arrivals = obaUserClient.getArrivalsAndDeparturesForStop(userData.getStopId());
+        StringBuilder sb = new StringBuilder();
+        for (ObaArrivalInfo obaArrival: arrivals) {
+            log.info("Arrival: " + obaArrival);
+            ArrivalInfo arrival = new ArrivalInfo(obaArrival, Instant.now().toEpochMilli());
```

Comment:
re: clock sync - `.getCurrentTime()` is a method of [`ObaArrivalInfoResponse`](https://github.com/OneBusAway/onebusaway-client-library/blob/master/src/main/java/org/onebusaway/io/client/request/ObaArrivalInfoResponse.java), which isn't directly exposed with the current setup - we'd need to structure the `obaUserClient` class to return the `ObaArrivalInfoResponse` instead of the `ObaArrivalInfo[]`, so we can pass in `ObaArrivalInfoResponse.getCurrentTime()` to the `ArrivalInfo` constructor.  I can refactor this once the PR is merged.